cypress/e2e/buy-shirt.cy.ts

```diff
@@ -1,29 +1,48 @@
+import {
+  AddressStepPage,
+  LoginPage,
+  MenuContentPage,
+  PaymentStepPage,
+  ProductsListPage,
+  ShippingStepPage,
+  ShoppingCartPage,
+} from "../page/index";
+
+const menuContentPage = new MenuContentPage();
+const productsListPage = new ProductsListPage();
+const shoppingCartPage = new ShoppingCartPage();
+const loginPage = new LoginPage();
+const addressStepPage = new AddressStepPage();
+const shippingStepPage = new ShippingStepPage();
+const paymentStepPage = new PaymentStepPage();
+
+
 describe("Buy a t-shirt", () => {
   it("then the t-shirt should be bought", () => {
-    cy.visit("http://automationpractice.com/");
-    cy.get("#block_top_menu > ul > li:nth-child(3) > a").click();
-    cy.get(
-        "#center_column a.button.ajax_add_to_cart_button.btn.btn-default",
-    ).click();
-    cy.get("[style*='display: block;'] .button-container > a").click();
-    cy.get(".cart_navigation span").click();
+    menuContentPage.visitMenuContentPage();
+    menuContentPage.goToTShirtMenu();
+
+    productsListPage.addTShirtToCart();
 
-    cy.get("#email").type("aperdomobo@gmail.com");
-    cy.get("#passwd").type("WorkshopProtractor");
-    cy.get("#SubmitLogin").click();
+    shoppingCartPage.proceedToCheckout();
+    shoppingCartPage.proceedToCheckoutSummary();
 
-    cy.get(".cart_navigation > .button > span").click();
+    loginPage.typeEmail("aperdomobo@gmail.com");
+    loginPage.typePassword("WorkshopProtractor");
+    loginPage.clickOnSignIn();
```

Comment:
Thinking about the action that is behind these three methods, what we want is to login or signIn, so, we can simplify this, having just one method called login or signIn; email and password should be parameters of that method.

Reply:
Resolved!